tests/bulk_create/tests.py

```diff
@@ -322,6 +328,129 @@ def test_ignore_conflicts_ignore(self):
+        self.assertEqual(UniqueTwo.objects.count(), 2)
+        new_objects = [
+            UniqueTwo(unique1=1, unique2=1, will_update=True),
+            UniqueTwo(unique1=2, unique2=2, will_update=True),
+        ]
+
+        UniqueTwo.objects.bulk_create(
+            new_objects, update_conflicts=True,
+            unique_fields=['unique1'], update_fields=['will_update']
+        )
+        self.assertEqual(UniqueTwo.objects.count(), 2)
+        # new objs
+        for obj in new_objects:
+            self.assertIsNone(obj.pk)
+            need_check = UniqueTwo.objects.get(
+                unique1=obj.unique1, unique2=obj.unique2
+            )
+            self.assertEqual(need_check.will_update, obj.will_update)
+
+        new_objects_2 = [
+            UniqueTwo(unique1=3, unique2=1, will_update=True),
+            UniqueTwo(unique1=4, unique2=2, will_update=True),
+        ]
+
+        UniqueTwo.objects.bulk_create(
+            new_objects_2, update_conflicts=True,
+            unique_fields=['unique2'], update_fields=['will_update']
+        )
+        self.assertEqual(UniqueTwo.objects.count(), 2)
+        # new objs
+        for obj in new_objects:
+            self.assertIsNone(obj.pk)
+            need_check = UniqueTwo.objects.get(
+                unique1=obj.unique1, unique2=obj.unique2
+            )
+            self.assertEqual(need_check.will_update, obj.will_update)
+
+    @skipUnlessDBFeature('supports_update_conflicts', 'supports_update_conflicts_with_target')
+    def test_update_unique_two_fields_error(self):
+        data = [
+            UniqueTwo(unique1=1, unique2=1, will_update=False),
+            UniqueTwo(unique1=2, unique2=2, will_update=False),
+        ]
+        UniqueTwo.objects.bulk_create(data)
+        self.assertEqual(UniqueTwo.objects.count(), 2)
+        new_objects = [
+            UniqueTwo(unique1=1, unique2=1, will_update=True),
+            UniqueTwo(unique1=2, unique2=2, will_update=True),
+        ]
+
+        with self.assertRaises((OperationalError, ProgrammingError), ):
+            UniqueTwo.objects.bulk_create(
+                new_objects, update_conflicts=True,
+                unique_fields=['unique1', 'unique2'], update_fields=['will_update']
+            )
+
+    @skipIfDBFeature('supports_update_conflicts_with_target')
+    @skipUnlessDBFeature('supports_update_conflicts')
+    def test_bulk_create_update_fields_error_without_unique(self):
+        # Using update_conflict but without update_fields, there's a problem.
+        msg = 'You need to specify which fields you want to update.'
+        with self.assertRaisesMessage(ValueError, msg):
+            TwoFields.objects.bulk_create(self.data, update_conflicts=True)
+
+        # Will not support update with unique fields.
+        msg = ('This database backend does not support updating '
+               'conflicts with specifying unique fields that will '
+               'trigger the upsert.')
```

Comment:
Use hanging indentation as described in [Python coding style](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style):
```suggestion
        msg = (
            'This database backend does not support updating conflicts with '
            'specifying unique fields that will trigger the upsert.'
        )
```